### PR TITLE
applications: asset_tracker_v2: thingy91x support

### PR DESCRIPTION
**Pull request by maxd-nordic**
This patch adds support for the Thingy:91 X
to the Asset Tracker V2.

### CODE REVIEW

File: applications/asset_tracker_v2/boards/thingy91x_nrf9151_ns.conf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+CONFIG_PWM=y
+CONFIG_LED_GPIO=y
+
+CONFIG_PM_PARTITION_REGION_SETTINGS_STORAGE_EXTERNAL=y
+
+CONFIG_ADXL367=y
+CONFIG_ADXL367_TRIGGER_GLOBAL_THREAD=y
+CONFIG_ADXL367_REFERENCED_ACTIVITY_DETECTION_MODE=y
+CONFIG_ADXL367_ACTIVITY_THRESHOLD=1000
+CONFIG_ADXL367_ACTIVITY_TIME=0
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/boards/thingy91x_nrf9151_ns.conf:27]: **suggestion:** explain these values

File: applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ */
+
+/* This structure enforces the header file to be included only once in the build.
+ * Violating this requirement triggers a multiple definition error at link time.
+ */
+const struct {} led_state_def_include_once;
+
+enum led_id {
+	LED_ID_1,
+
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h:21]: **nitpick:** 
```suggestion
```

File: applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	LED_ID_COUNT,
+
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h:23]: **nitpick:**
```suggestion
```

File: applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	[LED_STATE_ACTIVE_MODE]		= LED_EFFECT_LED_BREATH(LED_PERIOD_NORMAL,
+								LED_COLOR(0, 255, 255)),
+	[LED_STATE_PASSIVE_MODE]	= LED_EFFECT_LED_BREATH(LED_PERIOD_NORMAL,
+								LED_COLOR(0, 0, 255)),
+	[LED_STATE_ERROR_SYSTEM_FAULT]	= LED_EFFECT_LED_ON(LED_COLOR(255, 0, 0)),
+	[LED_STATE_FOTA_UPDATING]	= LED_EFFECT_LED_BREATH(LED_PERIOD_RAPID,
+								LED_COLOR(255, 165, 0)),
+	[LED_STATE_FOTA_UPDATE_REBOOT]	= LED_EFFECT_LED_ON(LED_COLOR(255, 165, 0)),
+	[LED_STATE_TURN_OFF]		= LED_EFFECT_LED_OFF(),
+};
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h:65]: **nitpick:** align whitespace in code block

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 5]
```diff
@@ -391,8 +421,15 @@ int ext_sensors_air_quality_get(uint16_t *ext_bsec_air_quality)
 
 int ext_sensors_accelerometer_threshold_set(double threshold, bool upper)
 {
+#if defined(CONFIG_ADXL367)
+	if (!upper) {
+		/* cannot distinguish events on adxl367 yet */
+		return 0;
+	}
+#endif
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c:429]:
```suggestion
#endif /* defined(CONFIG_ADXL367) */
```

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 6]
```diff
@@ -435,22 +472,24 @@ int ext_sensors_accelerometer_threshold_set(double threshold, bool upper)
 		evt_handler(&evt);
 		return err;
 	}
+#endif /* defined(CONFIG_ADXL362) */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c:475]:
```suggestion
#endif /* defined(CONFIG_ADXL362) || defined(CONFIG_ADXL367) */
```

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 7]
```diff
@@ -468,29 +507,35 @@ int ext_sensors_inactivity_timeout_set(double inact_time)
...
 	}
 		return 0;
+#endif /* defined(CONFIG_ADXL362) */
+
 error:
 	LOG_ERR("Could not set trigger for device %s, error: %d",
 		accel_sensor_lp.dev->name, err);
 	evt.type = EXT_SENSOR_EVT_ACCELEROMETER_ERROR;
 	evt_handler(&evt);
+#endif /* defined(CONFIG_ADXL362) */
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c:539]:
```suggestion
#endif /* defined(CONFIG_ADXL362) || defined(CONFIG_ADXL367) */
```

**Review by coderbyheart — APPROVED** [review]

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 8]
```diff
@@ -20,45 +20,89 @@
...
+
+static struct sensor_trigger accel_lp_sensor_trigger_motion = {
+		.chan = SENSOR_CHAN_ACCEL_XYZ,
+		.type = SENSOR_TRIG_MOTION
+};
+
+static struct sensor_trigger accel_lp_sensor_trigger_stationary = {
+		.chan = SENSOR_CHAN_ACCEL_XYZ,
+		.type = SENSOR_TRIG_STATIONARY
+};
```
Comment (bjda) [inline applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c:77]: Indentation

File: applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c [hunk 9]
```diff
@@ -20,45 +20,89 @@
...
+/** Sensor struct for the low-power accelerometer */
+static struct env_sensor accel_sensor_lp = {
+	.channel = SENSOR_CHAN_ACCEL_XYZ,
+	.dev = DEVICE_DT_GET(DT_ALIAS(accelerometer)),
 };
 
+static struct sensor_trigger accel_lp_sensor_trigger_motion = {
+		.chan = SENSOR_CHAN_ACCEL_XYZ,
+		.type = SENSOR_TRIG_THRESHOLD
+};
```
Comment (bjda) [inline applications/asset_tracker_v2/src/ext_sensors/ext_sensors.c:91]: Also indentation

**Review by bjda — APPROVED** [review]
Loogs good to me. A cople of nits on indentation

File: applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h (hunk 4 above)
Comment (maxd-nordic) [reply applications/asset_tracker_v2/include/thingy91x_nrf9151_ns/led_state_def.h:65]: (these are aligned with tab width 8)

**Pull request merged**